src/packages/version-check/README.md

```diff
@@ -0,0 +1,168 @@
+
+On install, VC defaults the `latestVersion` and `latestVersionLogged` to `0.0.0`. This will always be below the first-run of the latest install.
+
+VC is specced to fail silently. If a request fails, returns invalid semver or any combination of current/latest version is not valid semver it will quit without informing the user. It is self-healing if the API incorrectly reports a version and is later corrected it will inform the user of the latest, correct, version.
+
+VC's relies on the same `conf` package as truffle to manage the data persistence across environments (Linux, Mac, Windows).
+
+VC relies on the `semver` package to perform `diff` and validation of semver.
+
+#### Usage
+
+VC is flexible in how it may be implemented
+
+```javascript
+const vc = new VersionCheck(currentVersionSemVer, [config], [logger]);
+```
+
+The constructor requires a `currentVersionSemVer` and accepts an optional config and logger function.
+
+```javascript
+vc
+  .init();
+  .log();
+```
+
+`init` will send the request to fetch and set the `latestVersion` on disk.
+`log` will perform the required notification checks and, if they pass, log the banner message to the user based on the current `latestVersion` on disk.
+
+`init` will not await the fetch, it fires the request and does not concern itself with the outcome or results. If the desired behavior is to wait for a response before logging to the user (forgoing the eventual consistency spec), VC can be used as such:
+
+```javascript
+await vc.getLatestVersion();
+vc.log();
+```
+
+This usage will perform the the same steps as using `init`, however the `latestVersion` stored on disk will be the version returned from the fetch. This could be desirable by applications that are not concerned with latency/server timeout/ or that have configured a low `ttl` value in the VC config.
+
+```javascript
+vc.getVersionMessage();
+```
+
+`getVersionMessage` returns a single string noting the `currentVersion` and `latestVersion` if a semver change is detected. This string is appended to the `detailedVersion` string in `@ganache/cli` to support the `ganache --version` command switch.
+
+#### Test
+
+From a develop environment, start ganache using:
+
+```bash
+$ export VERSION="1.2.3"; npm run start
+```
+
+##### Expected Output
+
+1.) No banner message is displayed (first run, assumes latest was installed)
+
+```bash
+~/proj/ganache (feat/version-check) » export VERSION="1.2.3"; npm run start                                                                                                                                                                                                                                  hayek@rothbard
+
+> root@ start /home/hayek/proj/ganache
```

Comment:
maybe something like

```suggestion
> root@ start /home/user/proj/ganache
```

or maybe just remove all of these debug/irrelevant lines? /shrug